theano/tensor/blas.py

```diff
@@ -304,14 +305,16 @@ def default_blas_ldflags():
                 ['-L%s' % l for l in blas_info['library_dirs']] +
                 ['-l%s' % l for l in blas_info['libraries']] +
                 [])
-            if GCC_compiler.try_flags(ret):
+            if GCC_compiler.try_flags(ret) and try_blas_flag():
```

Comment:
@nouiz , For ```blas```, we also check the execution while for other flags we only check the compilation. Maybe we can also check execution for other flags too?